tls/s2n_client_key_exchange.c

```diff
@@ -128,6 +128,36 @@ int s2n_kem_client_key_recv(struct s2n_connection *conn, struct s2n_blob *shared
     return 0;
 }
 
+int s2n_hybrid_client_key_recv(struct s2n_connection *conn, struct s2n_blob *combined_shared_key)
+{
+    struct s2n_stuffer *io = &conn->handshake.io;
+    const struct s2n_kex *kex = conn->secure.cipher_suite->key_exchange_alg;
+    const struct s2n_kex *hybrid_kex_0 = kex->hybrid[0];
+    const struct s2n_kex *hybrid_kex_1 = kex->hybrid[1];
+
+    /* Keep a copy to the start of the entire hybrid client key exchange message for the hybrid PRF */
+    struct s2n_blob *client_key_exchange_message = &conn->secure.client_key_exchange_message;
+    client_key_exchange_message->data = s2n_stuffer_raw_read(io, 0);
+    notnull_check(client_key_exchange_message->data);
+    const int start = io->read_cursor;
```

Comment:
Isn't this converting a uint32_t to an int?

Reply:
Agreed, if `io->read_cursor` is 2^31 or greater, `start` will overflow and become negative since it's signed. In practice it shouldn't be possible to reach this condition since 2GB is greater than the max TLS Record size, but we should still update `start` to be a `uint32_t` here and elsewhere. 

We also have a backlog issue to fail the build on all potentially unsafe integer conversions here: https://github.com/awslabs/s2n/issues/789

Reply:
Fixed.